src/main/java/runge/meme/MemeController.java

```diff
@@ -0,0 +1,105 @@
+package runge.meme;
+
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.swing.*;
+import java.awt.*;
+import java.net.URL;
+
+public class MemeController
+{
+    private final JLabel imageLabel;
+    private final ImageService service;
+    private final DefaultListModel<String> listModel;
+
+    @Inject
+    public MemeController(
+        ImageService service,
+        @Named("listModel") DefaultListModel<String> listModel,
+        @Named("imageLabel")JLabel imageLabel
+        )
+    {
+        this.imageLabel = imageLabel;
+        this.listModel = listModel;
+        this.service = service;
+    }
+
+    public void addAndUpdateImage(String imageName) {
+        Disposable disposable = service.getMemeImage()
+                .subscribeOn(Schedulers.io())
+                .observeOn(SwingSchedulers.edt())
+                .subscribe(
+                        memeList -> {
+                            Memes[] getInfo = memeList.getData().getMemes();
+
+                            for (int i = 0; i < getInfo.length; i++)
+                            {
+                                listModel.addElement(getInfo[i].getName());
+                            }
+
+                            for (int i = 0; i < getInfo.length; i++)
```

Comment:
Why are these two loops?

Reply:
because one is to add all the names to the list and the other is to find the name clicked and get its image

Reply:
Why can't they be in the same loop?

Reply:
cause when I tried it didn't work?

Reply:
What happened?

Reply:
it only displayed the first name

Reply:
Oh because of the break.
Don't put it in the if statement.